dotCMS/src/main/java/com/dotcms/util/HttpRequestDataUtil.java

```diff
@@ -209,4 +218,29 @@ public static String getHostname(final HttpServletRequest request) {
 		return hostName;
 	}
 
+	/**
+	 * Convenience method to find out the server port and cache it to use it must likely in logs.
+	 *
+	 * @return String representing the server port
+	 */
+    public static Optional<String> getServerPort() {
+    	if (serverPort == null) {
+			final MBeanServer beanServer = ManagementFactory.getPlatformMBeanServer();
+			Set<ObjectName> objectNames;
+			String port;
+			try {
+				objectNames = beanServer.queryNames(
+						new ObjectName("*:type=Connector,*"),
+						Query.match(Query.attr("protocol"), Query.value("HTTP/1.1")));
+				port = objectNames.iterator().next().getKeyProperty("port");
+			} catch (MalformedObjectNameException e) {
+				port = null;
```

Comment:
Codacy found an issue: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)